packages/core/src/types/asyncapi.ts

```diff
@@ -0,0 +1,1141 @@
+    description: { type: 'string' },
+    externalDocs: 'ExternalDocs',
+  },
+  required: ['name'],
+};
+
+const ExternalDocs: NodeType = {
+  properties: {
+    description: { type: 'string' },
+    url: { type: 'string' },
+  },
+  required: ['url'],
+};
+
+const SecurityRequirement: NodeType = {
+  // done
+  properties: {},
+  additionalProperties: { type: 'array', items: { type: 'string' } },
+};
+
+const ServerBindings: NodeType = {
+  properties: {},
+  allowed() {
+    // allow all supported values, not all have deep linting
+    return [
+      'http',
+      'ws',
+      'kafka',
+      'anypointmq',
+      'amqp',
+      'amqp1',
+      'mqtt',
+      'mqtt5',
+      'nats',
+      'jms',
+      'sns',
+      'solace',
+      'sqs',
+      'stomp',
+      'redis',
+      'mercure',
+      'ibmmq',
+      'googlepubsub',
+      'pulsar',
+    ];
+  },
+  additionalProperties: { type: 'object' },
+};
+
+const Server: NodeType = {
+  properties: {
+    url: { type: 'string' },
+    protocol: { type: 'string' },
+    protocolVersion: { type: 'string' },
+    description: { type: 'string' },
+    variables: 'ServerVariablesMap',
+    security: 'SecurityRequirementList',
+    bindings: 'ServerBindings',
+    tags: 'TagList',
+  },
+  required: ['url', 'protocol'],
+};
+
+const ServerMap: NodeType = {
+  properties: {},
+  additionalProperties: (_value: any, key: string) =>
+    key.match(/^[A-Za-z0-9_\-]+$/) ? 'Server' : undefined,
+};
+
+const ServerVariable: NodeType = {
+  // done
+  properties: {
+    enum: {
+      type: 'array',
+      items: { type: 'string' },
+    },
+    default: { type: 'string' },
+    description: { type: 'string' },
+    examples: {
+      type: 'array',
+      items: { type: 'string' },
+    },
+  },
+  required: [],
+};
+
+const Info: NodeType = {
+  // done
+  properties: {
+    title: { type: 'string' },
+    version: { type: 'string' },
+    description: { type: 'string' },
+    termsOfService: { type: 'string' },
+    contact: 'Contact',
+    license: 'License',
+  },
+  required: ['title', 'version'],
+};
+
+const Contact: NodeType = {
+  // done
+  properties: {
+    name: { type: 'string' },
+    url: { type: 'string' },
+    email: { type: 'string' },
+  },
+};
+
+const License: NodeType = {
+  // done
+  properties: {
+    name: { type: 'string' },
+    url: { type: 'string' },
+  },
+  required: ['name'],
+};
+
+const Parameter: NodeType = {
+  properties: {
+    description: { type: 'string' },
+    schema: 'Schema',
+    location: { type: 'string' },
+  },
+};
+
+const CorrelationId: NodeType = {
+  properties: {
+    description: { type: 'string' },
+    location: { type: 'string' },
+  },
+  required: ['location'],
+};
+
+const Message: NodeType = {
+  properties: {
+    messageId: { type: 'string' },
+    headers: 'Schema',
+    payload: 'Schema', // TODO strictly this does not cover all cases
+    correlationId: 'CorrelationId',
+
+    schemaFormat: { type: 'string' }, // todo
```

Comment:
Please use the `TODO:` notation and describe what needs to be done here.

Reply:
Expanded this and a couple of other comments